mwclient/client.py

```diff
@@ -979,7 +993,16 @@ def ask(self, query, title=None):
         while offset is not None:
             results = self.raw_api('ask', query='{query}|offset={offset}'.format(
                 query=query, offset=offset), http_method='GET', **kwargs)
-
-            offset = results.get('query-continue-offset')
-            for key, value in results['query']['results'].iteritems():
-                yield {key: value}
+            if results.get('error'):
+                if self.handle_api_result(results):
+                    offset = None
+                    yield {}
+            elif results.get('query'):
```

Comment:
This line can also be removed. If there is no 'query' in the results, then something would be seriously wrong, but the current test would just push that under the carpet :)